google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -245,25 +258,6 @@ public void defaultQuery() throws Exception {
     assertEquals("bar", documents.next().get("foo"));
   }
 
-  @Test
-  public void queryForServerTimestamps() throws Exception {
-    DocumentReference documentReference = addDocument("serverTime", FieldValue.serverTimestamp());
-    DocumentSnapshot documentSnapshot = documentReference.get().get();
-    Timestamp serverTime = documentSnapshot.getTimestamp("serverTime");
-
-    if (TimeUnit.NANOSECONDS.toMicros(serverTime.getNanos()) > 0) {
-      // If serverTime has a microsecond component, this query produces no results as `getDate()`
-      // truncates to milliseconds.
-      Query query = randomColl.whereEqualTo("foo", documentSnapshot.getDate("serverTime"));
-      QuerySnapshot querySnapshot = query.get().get();
-      assertEquals(0, querySnapshot.size());
-    }
-
-    Query query = randomColl.whereEqualTo("serverTime", serverTime);
```

Comment:
Hmm, the new test doesn't involve `Query`, so it's not equivalent. This seemed like a valuable thing to check, so perhaps the new test should also have it?

Reply:
Sounds good. I added a new test case (`queryWithMicrosecondPrecision`).